### DISCUSSION TITLE
How to use moleculer on a Kubernetes cluster?

### DISCUSSION THREAD

**Discussion by 0x524c** [moleculerjs/moleculer#1205] [Q&A]
Hey guys,

I discovered the moleculer project just an hour ago and I'm blown away by the concept (I am understanding in parts). I would like to understand a point here where I still could not find an example to understand this content. Look at this link, please [link](https://moleculer.services/docs/0.14/clustering.html)

My question is: for example, in the case of Kubernetes working with microservices, do I need to create several small projects, where each project your set of pods, services, etc? or is it a single project that I must deploy in the N nodes of my cluster and the molecular distributes the load to different nodes that contain a set of services. In this case, all nodes would have a large replicated system, but each node would run a portion of the total available services?

Sorry for the long question, but I didn't fully understand the concept thinking about Kubernetes. I would like help to understand this. Thanks 😁

**Comment by 0x524c** [comment]
I reread the link I posted myself and paid more attention to this part "Just select a transporter, load one service per node and you’re done. Your application is running in microservices architecture".

This is exactly the scenario I'm in: I need to develop quickly, I can't spend a lot on a Kubernetes provider, but I also don't want to do everything monolithic and then have to redo the project to scale it in the future. The Moleculer is perfect for solving this, as I understand it.

So is that right? I can start as monolithic and then when I migrate to microservice I replicate my system and in N parts and each one would have a separate molecular service.

Is this how I would distribute load on my Kubernetes cluster then? 🤔😬

**Reply by shawnmcknight** [reply]
Today we build a single container image for all of our services and then create multiple deployments for each service, setting the environment variable to select which services are run for that deployment.

You could start as a monolith with separate services in the monolith on a single kubernetes deployment and then start separating them out later as your requirements grow

**Comment by 0x524c** [comment]
Hey @shawnmcknight 

"setting the environment variable to select which services run for this deployment." Thank you very much for answering. And I'm glad that they decided to do it this way, much easier so 🙏😁

**Comment by icebob** [comment]
Here you can find Kubernetes yamls of an example application: https://github.com/icebob/kantab/tree/master/kubernetes
If you see every Deployment uses the same docker image and uses `SERVICES` env var to configure which services should be loaded on the given pod. The load balancing is solved internally in the framework.

**Reply by 0x524c** [reply]
@icebob, thanks!!
